### PR TITLE
feat(platform): Add warning sonner toast for invalid otp

### LINKED ISSUE #332: PLATFORM: Show warning sonner toaster if OTP is invalid
## Description

In the current auth flow, there is no visual indicator for the user to know that the entered OTP is invalid. Therefore, we want to implement [`sonner`](https://ui.shadcn.com/docs/components/sonner)'s [Rich color warning](https://arc.net/l/quote/enrwcmur) state to prompt the user that the entered OTP is incorrect.

## Scope of Work
- Implement the [Rich color warning](https://arc.net/l/quote/enrwcmur) of [`sonner`](https://ui.shadcn.com/docs/components/sonner) in the `/auth/otp` page, if `/api/auth/validate-otp` is returning `401 Unauthorised`
- After you get the `401 Unauthorised` from the backend, set the `isLoading` state to false, so that the user can send the correct OTP next time

you can find the file to `/auth/otp` in `apps/platform/src/app/auth/otp/page.tsx`

### PR DESCRIPTION
**Pull request by PriyobrotoKar**
## Description

Added sonner rich text warning message for invalid otp

Fixes #332 

## Dependencies

[shadcn/ui-Sonner](https://ui.shadcn.com/docs/components/sonner)

## Future Improvements

_Mention any improvements to be done in future related to any file/feature_

## Mentions

_Mention and tag the people_

## Screenshots of relevant screens

Screenshot 2024-07-10 at 10 11 02 AM

## Developer's checklist

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-check on my work

**If changes are made in the code:**

- [x] I have followed the [coding guidelines](https://google.github.io/styleguide/jsguide.html)
- [x] My changes in code generate no new warnings
- [ ] My changes are breaking another fix/feature of the project
- [ ] I have added test cases to show that my feature works
- [x] I have added relevant screenshots in my PR
- [x] There are no UI/UX issues

### Documentation Update

- [ ] This PR requires an update to the documentation at [docs.keyshade.xyz](https://docs.keyshade.xyz)
- [x] I have made the necessary updates to the documentation, or no documentation changes are required.

___

### **PR Type**
Enhancement, Bug fix

___

### **Description**
- Added a warning toast message using `sonner` for invalid OTP input.
- Prevented default form submission behavior on OTP verification button click.
- Simplified the `setOtp` function call in the OTP input component.
- Added type definition for `OTPInputProps` in the `InputOTP` component.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>page.tsx</strong><dd><code>Add warning toast and improve OTP handling</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/platform/src/app/auth/otp/page.tsx

<li>Added <code>toast.warning</code> for invalid OTP input.<br> <li> Prevented default form submission on OTP verification.<br> <li> Simplified <code>setOtp</code> function call.<br>

</details>

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/335/files#diff-3ae286985cfeac090cabd429156a45eeaae5875dc9d9e23d93549e3b289387ba">+7/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>input-otp.tsx</strong><dd><code>Add type definition and simplify InputOTP component</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

apps/platform/src/components/ui/input-otp.tsx

<li>Added type definition for <code>OTPInputProps</code>.<br> <li> Simplified <code>InputOTP</code> component definition.<br>

</details>

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/335/files#diff-2905a64d2fff6a886f6db34c148dde4386dff9c2b056ae60ffb51cd7b84c32f7">+15/-13</a>&nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

**Review by kriptonian1 — APPROVED** [review]
LGTM 🚀

**Comment by kriptonian1** [comment]
@rajdip-b you can merge this

File: apps/platform/src/app/auth/otp/page.tsx [hunk 1]
```diff
@@ -61,6 +62,9 @@ export default function AuthOTPPage(): React.JSX.Element {
         }
       )
       if (response.status === 401) {
+        toast.warning(
+          'The OTP you entered is not valid, Please enter the right OTP'
```
[the remark this answers is not in the document]
Comment (PriyobrotoKar) [reply apps/platform/src/app/auth/otp/page.tsx:66]: Ok, so after I made this change and updated my branch with develop, I am getting this error while committing from prettier:
```
[error] No files matching the pattern were found: "packages/**/*.tsx".
ELIFECYCLE  Command failed with exit code 2.
```

[the remark this answers is not in the document]
Comment (rajdip-b) [reply apps/platform/src/app/auth/otp/page.tsx:66]: This is a problem with the prettier config. Can you replace `packages/**/*.{ts,tsx}` with `packages/**/*.ts,`?

https://github.com/PriyobrotoKar/keyshade/blob/4cf13c467c1eaa540ecddf44f1f6628ac8e783fd/package.json#L129

[the remark this answers is not in the document]
Comment (kriptonian1) [reply apps/platform/src/app/auth/otp/page.tsx:66]: wait prettier is not globally configured, wait let me have a look

[the remark this answers is not in the document]
Comment (kriptonian1) [reply apps/platform/src/app/auth/otp/page.tsx:66]: not sure why is tis happening

**Review by kriptonian1 — APPROVED** [review]
LGTM

**Review by rajdip-b — APPROVED** [review]

**Pull request merged**